src/material/list/selection-list.ts

```diff
@@ -579,7 +579,9 @@ export class MatSelectionList extends _MatSelectionListMixinBase implements CanD
 
   /** Returns the values of the selected options. */
   private _getSelectedOptionValues(): string[] {
-    return this.options.filter(option => option.selected).map(option => option.value);
+    return this.options.filter(option => {
+      return option.selected && option.value !== undefined;
+    }).map(option => option.value);
```

Comment:
should there also be a test that covers this change in terms of form control value. i.e. previously `undefined` was